### PR TITLE
Added cypress end to end tests

### PR DESCRIPTION
**Pull request by RayMoDev**

### CODE REVIEW

**Review by Darkweak — CHANGES_REQUESTED** [review]
Remove examples folder too

File: client/cypress.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "baseUrl": "http://localhost:4200",
+  "ignoreTestFiles": "**/examples/*",
```
Comment (Darkweak) [inline client/cypress.json:3]: Remove this line

File: client/cypress/fixtures/example.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Darkweak) [inline client/cypress/fixtures/example.json:1]: Useless file

File: client/cypress/integration/missions.spec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+describe('Should test missions', () => {
+  it('Should add, edit, delete a mission', () => {
+    cy.visit('http://localhost:4200/login');
+    cy.get('[data-cy=login]').type('jenaye');
+    cy.get('[data-cy=password]').type('jenaye');
+    cy.get('[data-cy=submit]').click();
```
Comment (Darkweak) [inline client/cypress/integration/missions.spec.js:6]: Use before to connect and store token in the storage to stop re-login user for each steps

File: client/src/app/components/login/login.component.html [hunk 4]
```diff
@@ -10,9 +10,9 @@
...
-                <input class="custom-input margin-bottom-40-px margin-top-40-px height-60-px width-100-prc padding-10-px font-input" placeholder="Password" type="password" name="password" [(ngModel)]="password">
-                <button class="vertical-align center title-text width-100-prc height-80-px rounded-light bg-purple-complement margin-bottom-40-px">Connection</button>
+                <input class="custom-input margin-bottom-10-px margin-top-40-px height-60-px width-100-prc padding-10-px font-input" type="text" name="login" [(ngModel)]="username" placeholder="Username" data-cy="login">
+                <input class="custom-input margin-bottom-40-px margin-top-40-px height-60-px width-100-prc padding-10-px font-input" placeholder="Password" type="password" name="password" [(ngModel)]="password" data-cy="password">
+                <button class="vertical-align center title-text width-100-prc height-80-px rounded-light bg-purple-complement margin-bottom-40-px" data-cy="password" data-cy="submit">Connection</button>
```
Comment (Darkweak) [inline client/src/app/components/login/login.component.html:15]:
```suggestion
                <button class="vertical-align center title-text width-100-prc height-80-px rounded-light bg-purple-complement margin-bottom-40-px" data-cy="submit">Connection</button>
```

**Review by Darkweak — CHANGES_REQUESTED** [review]

**Comment by RayMoDev** [comment]
@Darkweak  thank's i will update that

File: client/cypress/fixtures/data.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "username": "jenaye",
+  "password": "jenaye",
+  "baseUrl": "http://localhost:4200",
+  "backendUrl": "http://localhost:8000",
+  "dashboardRoute": "/dashboard",
+  "loginRoute": "/login",
+  "missionsRoute": "/missions",
+  "vulnerabilitiesRoute": "/vulnerabilities"
+}
```
Comment (Darkweak) [inline client/cypress/fixtures/data.json:10]: Add new line at the end of the file

**Review by Darkweak — APPROVED** [review]
Nice job, I approve this, just fix the last comment then LGTM 👍

File: client/cypress/fixtures/data.json (hunk 5 above)
Comment (RayMoDev) [reply client/cypress/fixtures/data.json:10]: @Darkweak  thank's , it will be merged or i  still have to make this tiny modification before ?

Comment (Darkweak) [reply client/cypress/fixtures/data.json:10]: We won't merge until the last fix is done because we don't want to increase the reminder/todo list :)

**Comment by jenaye** [comment]
Hello @RayMoDev 
Well done !  :100:   don't forget to add yourself in the readme in the contributors part :)

**Review by jenaye — APPROVED** [review]

File: client/cypress/fixtures/data.json (hunk 5 above)
Comment (RayMoDev) [reply client/cypress/fixtures/data.json]: @Darkweak Done !

**Comment by RayMoDev** [comment]
@jenaye Thank's for review ! i have done the last fix and added my self in the readme ;)

**Pull request merged**